### PR TITLE
Stronger language about Docker Compose customizability

### PR DESCRIPTION
**Pull request by potiuk**
Despite our warnings, our users continue treating the Docker
Compose that we exposed as something that should be easy to
extend and customize for their own needs, yet they continue
to struggle with some basic behaviour of containers, Docker Compose
and how they interact. This results in vast space of potential
problems as Docker Compose gives the user a false premise of
something that "just works" where it requires quite a deep
understanding on how it works.

When you get things wrong with Docker Compose, you often end up
with extremely confusing messages, that might suggest that the
problem is with Airflow, but really the problem is with how users
interact with their custom Docker images, registries, pulling,
networking, mounting volumes and plenty other things.

While this is the same with Kubernetes and Helm Chart, Helm Chart makes
it infinitely easier to customize in declarative way (this is what
our values.yaml does) and anything that has not been foreseen by Helm
Chart developers is "hard" by definition.

Docker Compose makes no such distinction. You really can't make Docker
Compose customizable by configuration, and any customization in it
requires modifying the compose file and for people who do not know
what they are doing will eventually lead to errors that they are not
able to diagnose and leads to creation of "Airlfow isssues", where they
should be brought to "Docker Compose" issues.

Example of that is here: https://github.com/apache/airflow/discussions/22301
where there are at least two issues that are not reproducible without
knowing in detail what the user has done, how the image was build
and distributed, and how the docker-compose installation interacted
with them. This leads to a terrible distraction for supporting
users of Airflow as the issues are really Docker Compose issues and
Airflow maintainers should not be involved in solving those.

This PR adds a bit stronger language and statement about the scope
and customizability of the Quick Start Docker Compose of ours. Not
only mentioning "Lack of Production Readiness" but also the
responsibility of the user to understand and diagnose docker compose
errors on their own and setting expectations that issues with Docker
Compose running should be directed elsewhere.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Hey everyone - here is my take on stronger language that we should IMHO put on the Docker-Compose Quick Start. 
I've been invovled in a number of issues raised by our users and honestly I think the "premise" of docker-compose that you can "just use it" and modify and you will be good, without really understanding of the "intricacies" of it does much more harm than good - giving the users false expectations on "how easy" it is to get airflow running with docker-compose. 

Yeah. Easy to start, Extremely difficult to diagnose and solve problems if you do not know exactly what you are doing. I think we should be very explicit that if user chooses docker-compose based on our quick start - the moment they touch the file, they are mostly on their own.

I'd love to hear other's thoughts about it, but this is the experience I have with users who think that our "docker-compose" is on our website, means that we will help them to solve their "learn how docker-compose works" problems.

Many times the comments I saw from the users was `But I am using the "official" docker-compose. It shoudl work!". I think this is false expectations and this PR is another attempt to be even clearer about the expectations users should have.

File: docs/apache-airflow/start/docker.rst [hunk 1]
```diff
@@ -22,6 +22,26 @@ Running Airflow in Docker
...
+==========================================
+
+DO NOT attempt to customize images and the Docker Compose if you do not know exactly what you are doing,
+know Docker Compose or do not will to debug and resolve problems on your own. If you do not
+know Docker Compose and expect Airflow to **just work** beyond following precisely the quick-start,
+then please use other ways of running Airflow - for example :doc:`/start/local` for testing and trying
+and :doc:`Official Airflow Community Helm Chart<helm-chart:index>` for production purposes.
+
+Even if many users think of Docker Compose as "ready to use", it is really a developer tool, that requires
+the user to know very well how docker images, containers docker compose networking, volumes, naming, image
```
Comment (potiuk) [inline docs/apache-airflow/start/docker.rst]:
```suggestion
the user to know very well how docker images, containers, docker compose networking, volumes, naming, image
```

**Comment by potiuk** [comment]
Another example https://github.com/apache/airflow/discussions/22322. Users have much bigger expectations when seeing our docker compose and I think we should be more explicit in telling them their expectations cannot be fulfilled

File: docs/apache-airflow/start/docker.rst [hunk 2]
```diff
@@ -22,6 +22,26 @@ Running Airflow in Docker
 
 This quick-start guide will allow you to quickly start Airflow with :doc:`CeleryExecutor </executor/celery>` in Docker. This is the fastest way to start Airflow.
 
+Customizing the quick-start Docker Compose
+==========================================
+
+DO NOT attempt to customize images and the Docker Compose if you do not know exactly what you are doing,
+know Docker Compose or do not will to debug and resolve problems on your own. If you do not
```
Comment (dstandish) [inline docs/apache-airflow/start/docker.rst]:
```suggestion
do not know Docker Compose, or are not prepared to debug and resolve problems on your own. If you do not
```

File: docs/apache-airflow/start/docker.rst [hunk 3]
```diff
@@ -22,6 +22,26 @@ Running Airflow in Docker
 
 This quick-start guide will allow you to quickly start Airflow with :doc:`CeleryExecutor </executor/celery>` in Docker. This is the fastest way to start Airflow.
```
Comment (dstandish) [inline docs/apache-airflow/start/docker.rst]: If docker compose is really intended as a dev tool, maybe this line should be reworked to reflect that.  I'm not sure exactly what the language should be.  But maybe something like

```suggestion
This quick-start guide will allow you to quickly get Airflow up and running with :doc:`CeleryExecutor </executor/celery>` in Docker.  For running airflow in production, we recommend running on kubernetes with the official helm chart.
```

**Review by dstandish — APPROVED** [review]

**Pull request merged**